crates/editor/src/editor.rs

```diff
@@ -7287,17 +7287,58 @@ impl Editor {
         let buffer = self.buffer.read(cx);
         let head = self.selections.newest::<usize>(cx).head();
         let (buffer, head) = buffer.text_anchor_for_position(head, cx)?;
+        let head_offset = text::ToOffset::to_offset(&head, buffer.read(cx));
         let replica_id = self.replica_id(cx);
 
         let workspace = self.workspace()?;
         let project = workspace.read(cx).project().clone();
         let references = project.update(cx, |project, cx| project.references(&buffer, head, cx));
-        Some(cx.spawn(|_, mut cx| async move {
-            let locations = references.await?;
+        Some(cx.spawn(|editor, mut cx| async move {
+            let mut locations = references.await?;
+
+            // Lsp may return references that contain self pos (eg. rust-analyzer)
+            // So we remove self pos from locations
```

Comment:
Can you make the comment a little bit clearer? What's a `self pos`, for example? The definition itself?

Reply:
Sure!